substrate/frame/beefy/src/equivocation.rs

```diff
@@ -122,19 +124,34 @@ where
 pub struct EquivocationReportSystem<T, R, P, L>(sp_std::marker::PhantomData<(T, R, P, L)>);
 
 /// Equivocation evidence convenience alias.
-pub type EquivocationEvidenceFor<T> = (
-	EquivocationProof<
-		BlockNumberFor<T>,
-		<T as Config>::BeefyId,
-		<<T as Config>::BeefyId as RuntimeAppPublic>::Signature,
-	>,
-	<T as Config>::KeyOwnerProof,
-);
+pub enum EquivocationEvidenceFor<T: Config + pallet_mmr::Config> {
+	VoteEquivocationProof(
+		VoteEquivocationProof<
+			BlockNumberFor<T>,
+			<T as Config>::BeefyId,
+			<<T as Config>::BeefyId as RuntimeAppPublic>::Signature,
+		>,
+		<T as Config>::KeyOwnerProof,
+	),
+	ForkEquivocationProof(
+		ForkEquivocationProof<
+			BlockNumberFor<T>,
+			<T as Config>::BeefyId,
+			<<T as Config>::BeefyId as RuntimeAppPublic>::Signature,
+			HeaderFor<T>,
+			<<T as pallet_mmr::Config>::Hashing as sp_runtime::traits::Hash>::Output,
```

Comment:
Yes, my suggestion was just because `<<T as pallet_mmr::Config>::Hashing as sp_runtime::traits::Hash>::Output` was too verbose. If it's less verbose now, we can leave it as it is.